packages/desktop-client/src/components/reports/reports/Spending.tsx

```diff
@@ -156,169 +181,232 @@ export function Spending() {
+                </Button>
+
+                <View
+                  style={{
+                    width: 1,
+                    height: 28,
+                    backgroundColor: theme.pillBorderDark,
+                    marginRight: 10,
+                    marginLeft: 10,
+                  }}
+                />
+              </>
+            )}
+
+            <View
               style={{
+                alignItems: 'center',
+                flexDirection: 'row',
+                marginRight: 5,
+                gap: 5,
+              }}
+            >
+              <Text>
+                <Trans>Compare</Trans>
+              </Text>
+              <Select
+                value={compare}
+                onChange={setCompare}
+                options={allIntervals.map(
+                  ({ name, pretty }) => [name, pretty] as const,
+                )}
+                style={{ width: 150 }}
+                popoverStyle={{ width: 150 }}
+              />
+              <Text>
+                <Trans>to</Trans>
+              </Text>
+              <Select
+                value={reportMode === 'single-month' ? compareTo : 'label'}
+                onChange={setCompareTo}
+                options={
+                  reportMode === 'single-month'
+                    ? allIntervals.map(({ name, pretty }) => [name, pretty])
+                    : [
+                        [
+                          'label',
+                          reportMode === 'budget'
+                            ? t('Budgeted')
+                            : t('Average spent'),
+                        ],
+                      ]
+                }
+                disabled={reportMode !== 'single-month'}
+                style={{ width: 150 }}
+                popoverStyle={{ width: 150 }}
+              />
+            </View>
+
+            <View
+              style={{
+                width: 1,
+                height: 28,
+                backgroundColor: theme.pillBorderDark,
+                marginRight: 15,
                 marginLeft: 10,
               }}
-              onPress={saveFilter}
-              isDisabled={filterSaved}
+            />
+
+            <View
+              style={{
+                flexDirection: 'row',
+                marginRight: 5,
+              }}
             >
-              {filterSaved ? 'Saved' : 'Save'}
-            </Button>
-          </Tooltip>
-        </View>
+              <ModeButton
+                selected={reportMode === 'single-month'}
+                style={{
+                  backgroundColor: 'inherit',
+                }}
+                onSelect={() => {
+                  setReportMode('single-month');
+                }}
+              >
+                <Trans>Single month</Trans>
+              </ModeButton>
+              <ModeButton
+                selected={reportMode === 'budget'}
+                onSelect={() => {
+                  setCompare(monthUtils.currentMonth());
```

Comment:
I'm not sure this is needed. I think it unnecessarily makes the UI a bit jumpy. Do we need this here?